api/repositories/security_group_repository.go

```diff
@@ -0,0 +1,101 @@
+package repositories
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"code.cloudfoundry.org/korifi/api/authorization"
+	apierrors "code.cloudfoundry.org/korifi/api/errors"
+	korifiv1alpha1 "code.cloudfoundry.org/korifi/controllers/api/v1alpha1"
+	"github.com/google/uuid"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const SecurityGroupResourceType = "Security Group"
+
+type SecurityGroupRepo struct {
+	userClientFactory authorization.UserClientFactory
+	rootNamespace     string
+}
+
+func NewSecurityGroupRepo(
+	userClientFactory authorization.UserClientFactory,
+	rootNamespace string,
+) *SecurityGroupRepo {
+	return &SecurityGroupRepo{
+		userClientFactory: userClientFactory,
+		rootNamespace:     rootNamespace,
+	}
+}
+
+type CreateSecurityGroupMessage struct {
+	DisplayName     string
+	Rules           []korifiv1alpha1.SecurityGroupRule
```

Comment:
Again, we should not be using korifiv1alpha1 in the message or record types (see above) :(

They should be modelled in the `repository` package and the `Create` method should convert korifiv1lpha1 types to record ones ones